pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -179,6 +190,22 @@ func (p *Provisioner) Remove(cluster *v3.Cluster) (runtime.Object, error) {
 }
 
 func (p *Provisioner) Updated(cluster *v3.Cluster) (runtime.Object, error) {
+	var err error
+
+	if shouldCleanHelmReleases(features.HelmCleanupOnUpgrade.Enabled(), cluster) {
```

Comment:
Same concern I raised in an above thread. I don't think it makes sense for this action to happen exactly once in a cluster since that creates a solution for just this problem, which defeats the point of a controller IMO.

For example, if you upgrade to 1.25 -> 1.26 and there's a new API deprecated and removed that we want to add this logic too (i.e. myresource.cattle.io); if clusters have already run this to upgrade from 1.24 -> 1.25, they will all have a true condition, but they **should** run it again on all those clusters on upgrading from 1.25 -> 1.26 right? Otherwise, their Helm releases have only removed PSPs, not my new resource.

This also brings back the same question as above regarding what if a cluster has been "migrated"; if you have future changes to that map that we defined, it wouldn't make sense w.r.t. the condition.